cmd/zpool/zpool_main.c

```diff
@@ -2471,61 +2474,587 @@ zpool_do_import(int argc, char **argv)
 }
 
 typedef struct iostat_cbdata {
-	boolean_t cb_verbose;
+	uint64_t cb_flags;
 	int cb_name_flags;
 	int cb_namewidth;
 	int cb_iteration;
+	char **cb_vdev_names; /* Only show these vdevs */
+	unsigned int cb_vdev_names_count;
+	boolean_t cb_verbose;
+	boolean_t cb_literal;
+	boolean_t cb_scripted;
 	zpool_list_t *cb_list;
 } iostat_cbdata_t;
 
+/*  iostat labels */
+static const char *iostat_top_labels[][5] = {
+	[VS_DEFAULT] = {"capacity", "operations", "bandwidth", NULL},
+	[VS_LATENCY] = {"total_wait", "disk_wait", NULL},
+	[VS_QUEUES] = {"sync_queue", "async_queue", NULL},
+	[VS_HISTO] = {"total", "disk", "sync_queue", "async_queue", NULL},
+};
+
+static const char *iostat_bottom_labels[][10] = {
+	[VS_DEFAULT] = {"alloc", "free", "read", "write", "read", "write",
+									NULL},
+	[VS_LATENCY] = {"read", "write", "read", "write", NULL},
+	[VS_QUEUES] = {"read", "write", "read", "write", "scrub", NULL},
+	[VS_HISTO] = {"read", "write", "read", "write", "read", "write", "read",
+			"write", "scrub", NULL},
+};
+
+/*
+ * Return the number of strings in a null-terminated string array.
+ * For example:
+ *
+ * 	const char foo[] = {"bar", "baz", NULL}
+ *
+ * returns 2
+ */
+static uint64_t
+str_array_len(const char *array[])
+{
+	uint64_t count = 0;
+	while (array[count])
+		count++;
+
+	return (count);
+}
+
+/*
+ * Return a default column width for default/latency/queue columns. This does
+ * not include histograms, which have their columns autosized.
+ */
+static unsigned int
+default_column_width(iostat_cbdata_t *cb, enum vdev_stat_flags type)
+{
+	unsigned long column_width = 5; /* Normal niceprint */
+	static unsigned long widths[] = {
+		/*
+		 * Choose some sane default column sizes for printing the
+		 * raw numbers.
+		 */
+		[VS_DEFAULT] = 15, /* 1PB capacity */
```

Comment:
I'm not so sure.  It's true that 1PB doesn't cover the worst case but it definitely covers the vast vast majority of pools.  The worst case here is that the headings don't line up perfectly but that probably isn't a big deal since this the expectation is that parsable data is being consumed by a script.